text/0058-standard-library.md

```diff
@@ -0,0 +1,287 @@
+when the user writes `A::B::C`, and suppose the active namespace is `NS`, that
+only refers to the type `A::B::C`, and not `NS::A::B::C` (even if `NS::A::B::C`
+exists).
+If my reading is correct, we also would want to extend this search behavior to
+namespaces, so that in the above case, in addition to looking for `A::B::C` and
+`NS::A::B::C` (probably in that order, for backwards compatibility), we would
+then look for `__cedar::A::B::C` if neither of the first two were found.
+
+### Alternative F: Envisioning multiple inheritance
+
+This RFC doesn't currently propose to allow multiple inheritance (one type
+extending multiple different types at once).
+Multiple inheritance would be straightforward in simple cases (just take the union
+of all attributes and parent types), but tricky in some edge cases (what happens
+if two base types declare the same attribute but with different types?).
+
+Multiple inheritance is an optional added feature; nothing in this RFC prevents
+us from introducing it in the future.
+However, in anticipation of a possible multiple-inheritance feature, we could
+require brackets around the `extends` list, like so:
+```
+entity MyEntityType extends [BaseEntityType] in ...
+```
+This would allow us to easily support multiple base entity types in the future,
+syntactically.
+
+However, even if we take this RFC as-is today, it would be perfectly reasonable
+to introduce multiple inheritance in the future and keep this RFC's syntax valid
+for single inheritance.
+The brackets would be optional for single inheritance and required for multiple
+inheritance.
+
+### Alternative G: Other ways to handle inheritance cycles
+
+This RFC proposes that we avoid inheritance cycles by requiring that the base
+type for `extends` must be a previously declared type (as in, declared farther
+up in the schema file).
+This would be the first schema feature for which the order of declarations in
+the schema file matters (as well as the order of schema files in
+[`Schema::from_schema_fragments()`]).
+An alternate design would be to simply disallow cycles, without introducing an
+order restriction.
+That is, if there is an inheritance cycle, the schema is illegal, but if there
+is not, the schema is allowed.
+
+This is strictly more permissive than the RFC's main proposal, so nothing
+prevents us from moving from the RFC's main proposal to this alternative in the
+future.
+
+I'm unclear which of these rules is easier to implement in practice, or which
+results in clearer errors for users.
+
+### Alternative H: Bikeshedding the `extends` syntax
+
+This RFC proposes that `extends` must appear before `in` and attributes:
+```
+entity MyEntityType extends A in [B];
+```
+We could instead require `extends` to appear _after_ `in` but before attributes:
+```
+entity MyEntityType in [B] extends A;
+```
+Or we could allow both.
+
+We could also consider some keyword other than `extends`, or some syntax other
```

Comment:
Downside of `extends` is that it implies object-oriented style inheritance. Which may be esp confusing since Cedar's type system doesn't have subtyping. I'll throw some other suggestions out:
`expands`
`copies`
`elaborates`

Reply:
`augments`

Reply:
These are good callouts.  I added a point about subtyping to Drawbacks, and I added these suggestions for alternate names for `extends` to Alternative H.